### PR TITLE
PCHR-2285: Turn Session into Model and refactor it

### PR DESCRIPTION
**Pull request by igorpavlov**
⚠️ This branch must be synced with 1.7 only when https://github.com/civicrm/civihr/pull/2038 is ready for new changes

## Overview

Session service now turns into Model since it will represent Session CiviHR Entity soon.

## Technical Details

1. Session moved into _models_ folder and named accordingly
2. Session now has its instance (along with basic tests)
3. Model tests **logic were not touched**, only dependencies injections were amended

## Comments

It was suggested that checked that 1.6 does not yet use Session anywhere.

image

---

- [x] Tests Pass

### CODE REVIEW

**Comment by igorpavlov** [comment]
I closed it accidentally.

File: org.civicrm.reqangular/src/common/models/instances/session-instance.instance.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/* eslint-env amd */
+
+define([
+  'common/modules/models-instances',
+  'common/models/instances/instance'
+], function (instances) {
+  'use strict';
+
+  instances.factory('SessionInstance', ['ModelInstance',
```
Comment (AkA84) [inline org.civicrm.reqangular/src/common/models/instances/session-instance.instance.js]: do we really need an instance in this case? the model simply returns a "static" object with some data in it

File: org.civicrm.reqangular/src/common/models/session.model.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  define([
+    'common/modules/models',
+    'common/models/model',
+    'common/models/instances/session-instance.instance'
+  ], function (models) {
+    'use strict';
+
+    models.factory('Session', [
+      '$log', '$q', 'Model', 'SessionInstance',
+      function ($log, $q, Model, SessionInstance) {
```
Comment (AkA84) [inline org.civicrm.reqangular/src/common/models/session.model.js]: you are not using the instance here

File: org.civicrm.reqangular/src/common/models/instances/session-instance.instance.js (hunk 1 above)
Comment (igorpavlov) [inline org.civicrm.reqangular/src/common/models/instances/session-instance.instance.js]: I made a placeholder. It is useless now, but in the past we had placeholders (ex. Absence-tab)

File: org.civicrm.reqangular/src/common/models/session.model.js (hunk 2 above)
Comment (igorpavlov) [inline org.civicrm.reqangular/src/common/models/session.model.js]: Will remove.

**Comment by igorpavlov** [comment]
Last commit includes built files and amends build files, no need to review.

**Pull request merged**